formula-coroutines/src/main/java/com/instacart/formula/coroutines/FlowRuntime.kt

```diff
@@ -27,35 +27,18 @@ object FlowRuntime {
         return callbackFlow<Output> {
             threadChecker.check("Need to subscribe on main thread.")
 
-            val mergedInspector = FormulaPlugins.inspector(
-                type = formula.type(),
-                local = inspector,
+            val runtime = FormulaRuntime(
+                threadChecker = threadChecker,
+                formula = formula,
+                onOutput = this::trySendBlocking,
+                onError = this::close,
+                inspector = inspector,
+                isValidationEnabled = isValidationEnabled,
             )
 
-            val runtimeFactory = {
-                FormulaRuntime(
-                    threadChecker = threadChecker,
-                    formula = formula,
-                    onOutput = this::trySendBlocking,
-                    onError = this::close,
-                    inspector = mergedInspector,
-                    isValidationEnabled = isValidationEnabled,
-                )
-            }
-
-            var runtime = runtimeFactory()
-
-            input.onEach { input ->
-                threadChecker.check("Input arrived on a wrong thread.")
-                if (!runtime.isKeyValid(input)) {
-                    runtime.terminate()
-                    runtime = runtimeFactory()
```

Comment:
It was a bit weird that we recreate the runtime if key changes. Instead, the runtime will handle state reset internally.